app/databaseAccessors/userAccessor.js

```diff
@@ -28,112 +28,110 @@ export default class UsersAccessor {
   }
 
   /**
-   * getUserByUsername Method
+   * Get User IDs by a email
    *
-   * This method retrieves the user MongoDB object from the
-   * database based on a given username
-   *
-   * @param {String} username
-   * @returns the User associated with the given username in
-   * the database.
+   * @param {String} email - Email
+   * @returns {ObjectId} - user's ID
    */
-  static async getUserByUsername(username) {
-    await Connection.open();
-    const user = await User.findOne({ username: username });
+  static async getUserIdByEmail(email) {
+    const user = await this.getUserByEmail(email);
+    if (!user) {
+      throw new ErrorUserNotFound(`User not found for email: ${email}`);
+    }
     return user;
   }
 
   /**
-   * Get User IDs by a list of usernames
-   *
-   * @param {Array<string>} usernames - List of usernames
-   * @returns {Array<ObjectId>} - List of user IDs
-   */
-  static async getUserIdsByUsernames(usernames) {
+ * Get a list of User IDs by a list emails
+ *
+ * @param {[String]} email - Array of email
+ * @returns {{[ObjectId]}} - Array of user ID
+ */
+  static async getUserIdsByMultipleEmails(emails) {
     const userIds = [];
-    for (const username of usernames) {
-      const user = await this.getUserByUsername(username);
+    for (const email of emails) {
+      const user = await this.getUserByEmail(email);
       if (user) {
         userIds.push(user._id);
       } else {
-        throw new ErrorUserNotFound(`User not found for username: ${username}`);
+        throw new ErrorUserNotFound(`User not found for email: ${email}`);
```

Comment:
(n-b) There should be a standardized way to tell whether or not an accessor should throw or if it returns null/empty for records. So people don't get caught off-guard of they expect a method like this to ignore emails of users that don't exist.

Reply:
In this case, it should return null/empty since that has been the practice for the other accessors; I will change it to reflect this. Should this become the standard?

Reply:
The use case of this particular method should throw, just noting that some people may want an additional method that returns null/empty. Can leave as is here